openfe_gromacs/tests/protocols/test_gromacs_md.py

```diff
@@ -198,6 +196,45 @@ def test_dry_many_molecules_solvent(
         )
 
 
+def test_gather(benzene_system, tmp_path_factory):
+    # check .gather behaves as expected
+    # Can't use mock.path since we need the outputs from the Setup Unit
+    # to execute the Run Unit
+    settings = GromacsMDProtocol.default_settings()
+    settings.output_settings_em.forcefield_cache = None
+    settings.simulation_settings_em.nsteps = 10
+    settings.simulation_settings_nvt.nsteps = 10
+    settings.simulation_settings_npt.nsteps = 1
+    settings.simulation_settings_em.rcoulomb = 1.0 * off_unit.nanometer
+    settings.simulation_settings_nvt.rcoulomb = 1.0 * off_unit.nanometer
+    settings.simulation_settings_npt.rcoulomb = 1.0 * off_unit.nanometer
+    protocol = GromacsMDProtocol(
+        settings=settings,
+    )
+
+    # create DAG from protocol and take first (and only) work unit from within
+    dag = protocol.create(
+        stateA=benzene_system,
+        stateB=benzene_system,
+        mapping=None,
+    )
+
+    shared_temp = tmp_path_factory.mktemp("shared")
+    scratch_temp = tmp_path_factory.mktemp("scratch")
+    dagres = gufe.protocols.execute_DAG(
+        dag,
+        shared_basedir=shared_temp,
+        scratch_basedir=scratch_temp,
+        keep_shared=False,
+        n_retries=3,
+    )
+    prot = GromacsMDProtocol(settings=settings)
+
+    res = prot.gather([dagres])
+
+    assert isinstance(res, GromacsMDProtocolResult)
```

Comment:
Maybe check the other attributes of the results object? I.e. what else do you expect to be getting out of it?

Reply:
Do you mean testing the `get_x` functions here?

Reply:
Yeah, it's a bit tedious but you can probably program/template a lot of it using something like getattr and a loop.